### PR TITLE
Add CircleCI  and Dockerfile

### PR DESCRIPTION
**Pull request by haroldwoo**
TODO:
- [x] Add creds to circleci
- [ ] Add branch merge protection requiring circleci build/test to pass
- [ ] Verify image got pushed to gcr

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-GitPython
-boto3
+GitPython==3.0.3
```
Comment (fbertsch) [inline requirements.txt]: Thank you for setting these explicitly!

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      - setup_remote_docker
+      - run: docker build -t app:build .
+
+  test:
+    docker:
+      - image: python:3.7
+    working_directory: ~/mozilla/probe-scraper
+    steps:
+      - checkout
+      - run: pip install -r requirements.txt -r test_requirements.txt
```
Comment (fbertsch) [inline .circleci/config.yml]: We definitely want these tests to run _inside_ the docker container. As such we'll need to add the `bin/entrypoint` file and run them using `docker run app:build test`.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      - run: pip install .
+      - run: flake8 --max-line-length 100 .
```
Comment (fbertsch) [inline .circleci/config.yml]: We can do the same for `flake8`

File: README.md [hunk 4]
```diff
@@ -72,11 +96,17 @@ These will not run by default, but will run on CI.
 Before opening a PR, it's good to test the code you wrote on the production data. You can specify a specific Firefox
 version to run on by using `first-version`:
 ```
+export COMMAND='python -m probe_scraper.runner --firefox-version 65 --dry-run'
+make run
+or locally via:
```
Comment (fbertsch) [inline README.md]: nit: This shouldn't be in code blocks

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+COPY . ${HOME}
+
+RUN pip install -r requirements.txt
+RUN pip install -r test_requirements.txt
+
+# Drop root and change ownership of the application folder to the user
+RUN chown -R ${USER_ID}:${GROUP_ID} ${HOME}
+USER ${USER_ID}
+
+CMD ["python", "-m", "probe_scraper.runner", "--glean", "--dry-run"]
```
Comment (fbertsch) [inline Dockerfile]: This is not the default `cmd` we want. I'm thinking we could leave out a default and just use `make run`.

**Comment by haroldwoo** [comment]
Yes it looks like I do have permissions to change settings on this repo

**Pull request merged**